dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/AbstractManagedObjectFactory.java

```diff
@@ -276,6 +267,18 @@ public CDIRuntime getCDIRuntime() {
         return mo;
     }
 
+    private String getBeanScope(boolean nonContextual) throws ManagedObjectException {
+        Bean<T> bean = null;
+        if (! nonContextual) {
+            bean = getBean();
+        }
+        String beanScope = null;
+        if (bean != null) {
+            beanScope = bean.getScope().getCanonicalName();
+        }
+        return beanScope;
+    }
```

Comment:
Needs comments but also how about structuring it like this...
```
    private String getBeanScope(boolean nonContextual) throws ManagedObjectException {
        String beanScope = null;
        if (! nonContextual) { //if nonContextual==true then there is no bean and no beanScope
            Bean<T> bean = getBean();
            if (bean != null) {
                beanScope = bean.getScope().getCanonicalName();
            }
        }
        return beanScope;
    }
```

Reply:
Sure, why not